Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/MaterialComponentController.cpp

```diff
@@ -192,39 +194,44 @@ namespace AZ
         {
             Data::AssetBus::MultiHandler::BusDisconnect();
 
-            bool anyQueued = false;
-            auto queueAsset = [&anyQueued, this](AZ::Data::Asset<AZ::RPI::MaterialAsset>& materialAsset) -> bool
+            // Build tables of all referenced materials so that we can look up default values for their material assets and properties
+            m_defaultMaterialMap.clear();
+            m_activeMaterialMap.clear();
+            m_uniqueMaterialMap.clear();
+            for (const auto& [materialAssignmentId, materialAssignment] : GetOriginalMaterialAssignments())
             {
-                if (materialAsset.GetId().IsValid() && !this->Data::AssetBus::MultiHandler::BusIsConnectedId(materialAsset.GetId()))
-                {
-                    anyQueued = true;
-                    materialAsset.QueueLoad();
-                    this->Data::AssetBus::MultiHandler::BusConnect(materialAsset.GetId());
-                    return true;
-                }
-                return false;
-            };
+                auto& defaultMaterial = m_defaultMaterialMap[materialAssignmentId];
+                defaultMaterial = materialAssignment.m_materialAsset;
 
-            for (auto& materialPair : m_configuration.m_materials)
-            {
-                if (materialPair.second.m_materialInstancePreCreated)
-                {
-                    continue;
-                }
+                auto& activeMaterial = m_activeMaterialMap[materialAssignmentId];
+                activeMaterial = defaultMaterial;
```

Comment:
I am confused as to what you're trying to do here, maybe you forgot to check if defaultMaterial/activeMaterial is invalid?